src/StatusQ/Components/StatusRoundedImage.qml

```diff
@@ -28,11 +28,18 @@ Rectangle {
         id: image
         width: statusRoundImage.width
         height: statusRoundImage.height
-        sourceSize.width: parent.width * 2
-        sourceSize.height: parent.height * 2
         fillMode: Image.PreserveAspectFit
         anchors.centerIn: parent
-        cache: true
+        cache: false
+
+        onSourceChanged: {
+            if (sourceSize.width < width || sourceSize.height < height) {
+                sourceSize = Qt.size(width * 2, height * 2)
+            }
+            else {
+                sourceSize = undefined
+            }
+        }
```

Comment:
It will not work properly when the image size is changed dynamically, because it's an imperative assignment, not a binding. You can use Qt.binding function to solve that.
```suggestion
            sourceSize = Qt.binding(function() {
                return sourceSize.width < width || sourceSize.height < height ? Qt.size(width * 2, height * 2) : undefined
            })
        }
```